api-put-object-common.go

```diff
@@ -67,20 +67,41 @@ func isReadAt(reader io.Reader) (ok bool) {
 //  minPartSize - 64MiB
 //  maxMultipartPutObjectSize - 5TiB
 //
-func optimalPartInfo(objectSize int64) (totalPartsCount int, partSize int64, lastPartSize int64, err error) {
+func optimalPartInfo(objectSize int64, configuredPartSize uint64) (totalPartsCount int, partSize int64, lastPartSize int64, err error) {
 	// object size is '-1' set it to 5TiB.
 	if objectSize == -1 {
 		objectSize = maxMultipartPutObjectSize
 	}
+
 	// object size is larger than supported maximum.
 	if objectSize > maxMultipartPutObjectSize {
 		err = ErrEntityTooLarge(objectSize, maxMultipartPutObjectSize, "", "")
 		return
 	}
+
+	if int64(configuredPartSize) > objectSize {
+		err = ErrEntityTooLarge(int64(configuredPartSize), objectSize, "", "")
+		return
+	}
+
+	if objectSize > (int64(configuredPartSize) * maxPartsCount) {
```

Comment:
this always returns an error for initial `objectSize = -1` and `configuredPartSize < ~524MB`, how can I upload objects of unknown size in smaller chunks?

Reply:
You cannot @bunin there is no such thing as unknown file size in AWS S3 protocol. We provide that as a special feature but the limitation is we stick to the maximum size possible to upload upto 5TiB.